src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
+                // match the element types in an array.
+                // Ex: string[,][] 
+                // [,] is a 2 dimension array with element type string[]
+                // [] is a 1 dimension array with element type string
+                foreach (var arrayExpression in arrayTypeNode.ArrayExpressions)
                 {
-                    if (document.FilePath == fileName)
+                    if (currentType is not IArrayTypeSymbol currentArrayType)
                     {
-                        return ImmutableArray.Create(document);
+                        return false;
                     }
 
-                    else if (document.Name == documentName)
+                    if (currentArrayType.Rank != arrayExpression.CommaTokens.Length + 1)
                     {
-                        potentialMatches.Add(document);
+                        return false;
                     }
+
+                    currentType = currentArrayType.ElementType;
+                }
+
+                // All array types have been exchausted from the
+                // stackframe identifier and the type is still an array
+                if (currentType is IArrayTypeSymbol)
+                {
+                    return false;
                 }
+
+                return MatchType(currentType, arrayTypeNode.TypeIdentifier);
             }
 
-            return potentialMatches.ToImmutableArray();
+            // Special types can have different casing representations
+            // Ex: string and String are the same (System.String)
+            var specialTypeName = TryGetSpecialTypeName(type.SpecialType);
+            if (specialTypeName is not null)
+            {
+                return specialTypeName == stackFrameType.ToString(skipTrivia: true);
+            }
+
+            // Default to just comparing the display name
+            return type.ToDisplayString() == stackFrameType.ToString(skipTrivia: true);
         }
+
+        private static string? TryGetSpecialTypeName(SpecialType type)
+            => type switch
+            {
+                SpecialType.System_Object => nameof(Object),
+                SpecialType.System_Boolean => nameof(Boolean),
+                SpecialType.System_SByte => nameof(SByte),
+                SpecialType.System_Byte => nameof(Byte),
+                SpecialType.System_Decimal => nameof(Decimal),
+                SpecialType.System_Single => nameof(Single),
+                SpecialType.System_Double => nameof(Double),
+                SpecialType.System_Int16 => nameof(Int16),
+                SpecialType.System_Int32 => nameof(Int32),
+                SpecialType.System_Int64 => nameof(Int64),
+                SpecialType.System_String => nameof(String),
+                SpecialType.System_UInt16 => nameof(UInt16),
+                SpecialType.System_UInt32 => nameof(UInt32),
+                SpecialType.System_UInt64 => nameof(UInt64),
```

Comment:
this is just equivalent to => `type.Name`  not sure why we need to actually do anything special here.  in stackframes i assume things just have their metadata name, not their C# primitive name.  So everything should just fall out.

Reply:
ah I see. I've been using `ToDisplayString()` which does do something special, and was trying to counteract that. I'll just change to comparing with Name if it's a special type.

Reply:
> I'll just change to comparing with Name if it's a special type.

I'd say you can also just always look at Name for all types.  And, if you want to use ToDisplayString, you can just provide an appropriate format specifier that says: don't use the speical types :)